src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1735,6 +1738,148 @@ GenTree* Lowering::LowerHWIntrinsicCreate(GenTreeHWIntrinsic* node)
     }
     else if (argCnt == 1)
     {
+        if (isCreateScalar)
+        {
+            switch (simdBaseType)
+            {
+                case TYP_BYTE:
+                case TYP_UBYTE:
+                {
+                    // Types need to be explicitly zero-extended to ensure upper-bits are zero
+                    // We need to explicitly use TYP_UBYTE since unsigned is ignored for small types
+                    // Explicitly handle both BYTE and UBYTE to account for reinterpret casts and the like
+
+                    tmp1 = comp->gtNewCastNode(TYP_UBYTE, op1, /* unsigned */ true, TYP_INT);
```

Comment:
So is it a correct way to zero-extend a small type? Isn't this creating a node with TYP_UBYTE in lower? cc @SingleAccretion. I though it supposed to be TYP_INT, op1, true, TYP_UBYTE

Reply:
True, that looks like a typo. Should be `gtNewCastNode(TYP_INT, op1, /* fromUnsigned */ false, TYP_UBYTE)` (`GTF_UNSIGNED` not set to not confuse people unnecessarily).

Reply:
Fixed. I really don't like how we handle casts for small types. It is especially painful for HWIntrinsics since we need to actually deal with small types and consume them directly as small types.